images/handlers.go

```diff
@@ -40,6 +40,10 @@ var (
 	// This applies only to a single descriptor in a handler
 	// chain and does not apply to descendant descriptors.
 	ErrStopHandler = fmt.Errorf("stop handler")
+
+	// ErrNoPlatformMatched is used when the platform matcher didn't
+	// match any of descriptor.
+	ErrEmptyWalk = fmt.Errorf("Empty Walk - image might be filtered out")
```

Comment:
not sure why the linter didn't pick this up, but the commented name (`ErrNoPlatformMatched`) does not equal `ErrEmptyWalk`; I also thought the linter complained when errors start with a capital letter.  Do we want to call it `NoPlatformMatch` as a clearer indicator of how this happened or do we foresee a case where an "empty walk" happens for other reasons?

Reply:
Indeed, it was initially named ``ErrNoPlatformMatched``, since I had a different implementation with a new platform handler, but that had some issues (some integration tests were failing), so I did this instead.

I don't remember exactly, but I think the linter wants these errors to be prefixed by ``Err``.

As for other "empty walks", ``FilterPlatforms`` seems to be the only filter defined in ``containerd/images/handlers.go``, but I haven't checked other instances of ``images.Walk``.

Also, keep in mind that this PR stacks 2 other PRs, which are mentioned in the PR description. They were needed in order for the CIs to pass with this test.